.github/workflows/docker.yml

```diff
@@ -0,0 +1,137 @@
+name: docker
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - 'master'
+
+jobs:
+  # Builds base image for stempy of various verions if Dockerfile.base has changed
+  build-stempy:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.9']
+        mpi: ['ON', 'OFF']
+        ipykernel: ['', 'ipykernel']
+        dev: ['', 'dev']
+        exclude:
+          - MPI: 'ON'
+            ipykernel: 'ipykernel'
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: 'recursive'
+
+      - name: 'Setup Docker Environment'
+        uses: ./.github/actions/docker_setup
+        with:
+          github_event_name: ${{ github.event_name }}
+          DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+          DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+          DOCKERHUB_ORG: ${{ vars.DOCKERHUB_ORG }}
+
+      - name: Determine if Dockerfile.base changed
+        id: changed-dockerfile-base
+        uses: tj-actions/changed-files@v35
+        with:
+          files: |
+            ./docker/Dockerfile.base
+            ./docker/apt-packages-common.txt
+            ./docker/apt-packages-dev.txt
```

Comment:
Does the `Dockerfile.base` take a long time to build? If not, it could have been simpler to just build it every time, rather than caching it. We may actually want to rebuild it occasionally just because upstream dependencies change over time which include things like security updates.

But we can leave this!

Reply:
Dockerfile.base does take a long time to build (~40 minutes). I understand that it is CI, but say you are developing on a fork and want to make a quick change and test at NERSC. 

All you have to do is edit this:
```yml
on:
  push:
    branches:
      - 'master'
  pull_request:
    branches:
      - 'master'
```

And commit your change, and CI will take care of building/pushing to your own Dockerhub repo (so long as you have set up these vars and secrets):
```
DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
DOCKERHUB_ORG: ${{ vars.DOCKERHUB_ORG }}
```

I can set up a cron job to build/push base, I don't think that would be too much work.

Reply:
Thanks for the info! It's probably okay for now unless @cjh1 thinks we should add a cron job.